solidity/contracts/converter/types/liquidity-pool-v2/LiquidityPoolV2Converter.sol

```diff
@@ -873,141 +674,21 @@ contract LiquidityPoolV2Converter is LiquidityPoolConverter {
-        if (referenceTime < _externalRateUpdateTime) {
-            referenceTime = _externalRateUpdateTime;
-        }
-
-        // limit the reference time by current time
-        uint256 currentTime = time();
-        if (referenceTime > currentTime) {
-            referenceTime = currentTime;
-        }
-
-        // if no time has passed since the reference time, return current weights (also ensures a single update per block)
-        uint256 elapsedTime = currentTime - referenceTime;
-        if (elapsedTime == 0) {
-            return (primaryReserveWeight, externalPrimaryReserveWeight);
-        }
-
-        // find the token whose weight is lower than the target weight and get its pool rate - if it's
-        // lower than external rate, update the weights
-        Fraction memory poolRate = tokensRate(
-            primaryReserveTokenLocal,
-            secondaryReserveToken,
-            primaryReserveWeight,
-            inverseWeight(primaryReserveWeight));
-
-        bool updateWeights = false;
-        if (primaryReserveWeight < externalPrimaryReserveWeight) {
-            updateWeights = compareRates(poolRate, _externalRate) < 0;
-        }
-        else {
-            updateWeights = compareRates(poolRate, _externalRate) > 0;
-        }
-
-        if (!updateWeights) {
-            return (primaryReserveWeight, externalPrimaryReserveWeight);
-        }
-
-        // if the elapsed time since the reference rate is equal or larger than the propagation time,
-        // the external rate should take full effect
-        if (elapsedTime >= externalRatePropagationTime) {
-            return (externalPrimaryReserveWeight, externalPrimaryReserveWeight);
-        }
-
-        // move the weights towards their target by the same proportion of elapsed time out of the rate propagation time
-        primaryReserveWeight = uint32(weightedAverageIntegers(
-            primaryReserveWeight, externalPrimaryReserveWeight,
-            elapsedTime, externalRatePropagationTime));
-        return (primaryReserveWeight, externalPrimaryReserveWeight);
-    }
-
-    /**
-      * @dev returns the current rate for add/remove liquidity rebalancing
-      * only used to circumvent the `stack too deep` compiler error
-      *
-      * @return effective rate
+      * @return rate
     */
-    function rebalanceRate() private view returns (Fraction memory) {
-        // if one of the balances is 0, return the external rate
-        if (reserves[primaryReserveToken].balance == 0 || reserves[secondaryReserveToken].balance == 0) {
-            Fraction memory externalRate;
-            (externalRate.n, externalRate.d) = priceOracle.latestRate(primaryReserveToken, secondaryReserveToken);
-            return externalRate;
-        }
-
-        // return the rate based on the current rate
-        return tokensRate(primaryReserveToken, secondaryReserveToken, 0, 0);
+    function _effectiveTokensRate() private view returns (Fraction memory) {

```

Comment:
we only use that function for converting between uint256,uint256 and Fraction, I wonder if we can make cleaner - maybe a utility function that does that?

Reply:
I don't understand this comment.

Reply:
I thought we could create a utility function that receives 2 uint256 values and returns a newly constructed Faction struct, and then get rid of `_effectiveTokensRate`, but some to think of it, I don't believe you can pass a tuple return value to function as argument

Reply:
I still don't understand.
We'd still need to call `priceOracle.latestRate(primaryReserveToken, secondaryReserveToken)`.
So what exactly are you suggesting (ideally, assuming that we don't have a compiler limitation)?

Reply:
was just looking for ways to remove that function that basically does nothing.